### PR TITLE
KAFKA-8507: Unify bootstrap-server flag for command line tools (KIP-499) part-2

### PR DESCRIPTION
**Pull request by stanislavkozlovski**
This patch updates ReplicaVerificationTool, TransactionalMessageCopier, GetOffsetShell and VerifiableLog4jAppender to add and prefer the --bootstrap-server flag for defining the connection point of the Kafka cluster. This change is part of KIP-499: https://cwiki.apache.org/confluence/display/KAFKA/KIP-499+-+Unify+connection+name+flag+for+command+line+tool.

### CODE REVIEW

**Comment by stanislavkozlovski** [comment]
This PR is a continuation of https://github.com/apache/kafka/pull/8023

**Comment by stanislavkozlovski** [comment]
System tests to run:
- replica_verification_tool.py

**Comment by stanislavkozlovski** [comment]
cc @hachikuji @chia7712 @rondagostino @mitchell-h

File: core/src/main/scala/kafka/tools/ConsoleProducer.scala [hunk 1]
```diff
@@ -125,7 +125,7 @@ object ConsoleProducer {
       .withRequiredArg
       .describedAs("topic")
       .ofType(classOf[String])
-    val brokerListOpt = parser.accepts("broker-list", "DEPRECATED, use --bootstrap-server instead; ignored if --bootstrap-server is specified.  The broker list string in the form HOST1:PORT1,HOST2:PORT2.")
+    val brokerListOpt = parser.accepts("broker-list", "DEPRECATED, use --bootstrap-server instead; ignored if --bootstrap-server is specified. The broker list string in the form HOST1:PORT1,HOST2:PORT2.")
```
Comment (hachikuji) [inline core/src/main/scala/kafka/tools/ConsoleProducer.scala:128]: The doc for `--bootstrap-server` below adds "REQUIRED unless --broker-list(deprecated) is specified." Can we drop that?

File: core/src/main/scala/kafka/tools/GetOffsetShell.scala [hunk 2]
```diff
@@ -32,66 +31,88 @@ import scala.collection.Seq
...
-                           .ofType(classOf[java.lang.Integer])
-                           .defaultsTo(1000)
+      .withRequiredArg
+      .describedAs("ms")
+      .ofType(classOf[java.lang.Integer])
+      .defaultsTo(1000)
+
+    options = parser.parse(args : _*)
+
+    def brokerList: String = {
```
Comment (hachikuji) [inline core/src/main/scala/kafka/tools/GetOffsetShell.scala]: nit: can we call this `bootstrapServers`? Similarly in other tools.

File: tests/kafkatest/services/performance/consumer_performance.py [hunk 3]
```diff
@@ -110,7 +110,7 @@ def args(self, version):
         if self.new_consumer:
             if version <= LATEST_0_10_0:
                 args['new-consumer'] = ""
-            args['broker-list'] = self.kafka.bootstrap_servers(self.security_config.security_protocol)
+            args['bootstrap-server'] = self.kafka.bootstrap_servers(self.security_config.security_protocol)
```
Comment (hachikuji) [inline tests/kafkatest/services/performance/consumer_performance.py]: Hmm.. The version check above suggests this tool still works with older versions. Not sure we'll be able to change these.

**Review by hachikuji — COMMENTED** [review]
Thanks, looks good overall. Left a few comments.

File: tests/kafkatest/services/performance/consumer_performance.py (hunk 3 above)
Comment (stanislavkozlovski) [reply tests/kafkatest/services/performance/consumer_performance.py]: Good catch

**Comment by stanislavkozlovski** [comment]
I started a run of the system tests in https://jenkins.confluent.io/job/system-test-kafka-branch-builder/3828/

**Comment by urbandan** [comment]
Hi, are there any plans on merging this PR?
I'm working on [KIP-635](https://cwiki.apache.org/confluence/display/KAFKA/KIP-635%3A+GetOffsetShell%3A+support+for+multiple+topics+and+consumer+configuration+override), and I could fix the broker-list arg for GetOffsetShell in that PR - but I would like to avoid potential conflicts.

**Pull request closed without merging**